autogen/agentchat/conversable_agent.py

```diff
@@ -597,6 +619,29 @@ def _prepare_chat(self, recipient, clear_history):
             self.clear_history(recipient)
             recipient.clear_history(self)
 
+    def _raise_exception_on_async_reply_functions(self) -> None:
+        """Raise an exception if any async reply functions are registered.
+
+        Raises:
+            RuntimeError: if any async reply functions are registered.
+        """
+        reply_functions = {f["reply_func"] for f in self._reply_func_list}.difference(
+            self._ignore_async_func_in_sync_chat_list
+        )
+
+        async_reply_functions = [f for f in reply_functions if asyncio.coroutines.iscoroutinefunction(f)]
+        if async_reply_functions != []:
+            msg = (
+                "Async reply functions can only be used with ConversableAgent.a_initiate_chat(). The following async reply functions are found: "
+                + ", ".join([f.__name__ for f in async_reply_functions])
+            )
+
+            raise RuntimeError(msg)
+
+        if hasattr(self, "_groupchat"):
```

Comment:
This opens up some interesting discussion about nesting as we might want to standardize the interface for nesting. I think this is fine for this PR for now.

Reply:
What about moving this part to groupchat.py and override this function in GroupChatManager?

Reply:
I am not sure how to do that. It has to work in chats without a GroupChatManager as well in cases where you only have a user proxy and a conversable agent.